src/Lexer.js

```diff
@@ -80,6 +80,8 @@ var tokenRegex = new RegExp(
     "([ \r\n\t]+)|" +                                 // whitespace
     "([!-\\[\\]-\u2027\u202A-\uD7FF\uF900-\uFFFF]" +  // single codepoint
     "|[\uD800-\uDBFF][\uDC00-\uDFFF]" +               // surrogate pair
+    "|\\\\verb\\*([^\0]).*?(?:\\3|\n|$)" +            // \verb*
+    "|\\\\verb([^*a-zA-Z]).*?(?:\\4|\n|$)" +          // \verb unstarred
```

Comment:
How about just writing `.*?\\4` (and likewise for `\\3`)? The use of `.` already excludes `\n`, as well as `\r`, `\u2028` and `\u2029`. I think that is a reasonable fit for how TeX does this, but `[\n]*?\\4` would be conceivable as well. If the terminating delimiter is *not* found before the end of line, then this would simply fall through to the function name getting matched against `\verb`, so we could define a function of that name to report the error.